openscvx/plotting/plotting.py

```diff
@@ -1,279 +1,298 @@
+        show: What to plot - "both", "nodes", or "trajectory"
 
-        # Add constraint bounds (hlines don't support legend, so we add invisible scatter traces)
-        if not np.isinf(x_min):
-            fig.add_hline(
-                y=x_min,
-                line={"color": "red", "width": 1, "dash": "dot"},
-                row=row,
-                col=col,
-            )
-        if not np.isinf(x_max):
-            fig.add_hline(
-                y=x_max,
-                line={"color": "red", "width": 1, "dash": "dot"},
-                row=row,
-                col=col,
-            )
+    Returns:
+        Plotly figure
+    """
+    available = {c.name for c in result._controls}
+    if control_name not in available:
+        raise ValueError(f"Control '{control_name}' not found. Available: {sorted(available)}")
+    if show not in ("both", "nodes", "trajectory"):
+        raise ValueError(f"show must be 'both', 'nodes', or 'trajectory', got '{show}'")
 
-        # Add bounds to legend (only once)
-        if show_legend and (not np.isinf(x_min) or not np.isinf(x_max)):
-            fig.add_trace(
-                go.Scatter(
-                    x=[None],
-                    y=[None],
-                    mode="lines",
-                    name="Bounds",
-                    showlegend=True,
-                    legendgroup="bounds",
-                    line={"color": "red", "width": 1, "dash": "dot"},
-                ),
-                row=row,
-                col=col,
-            )
+    dim = _get_var_dim(result, control_name, result._controls)
+    t_nodes = result.nodes["time"].flatten()
+    has_trajectory = bool(result.trajectory) and control_name in result.trajectory
+    t_full = result.trajectory["time"].flatten() if has_trajectory else None
 
-    # Update axis labels
-    for i in range(1, n_rows + 1):
-        fig.update_xaxes(title_text="Time (s)", row=i, col=1)
+    fig = go.Figure()
+    fig.update_layout(title_text=control_name, template="plotly_dark")
+
+    for i in range(dim):
+        label = f"{control_name}_{i}" if dim > 1 else control_name
+
+        if show in ("both", "trajectory") and has_trajectory:
+            data = result.trajectory[control_name]
+            y = data if data.ndim == 1 else data[:, i]
+            fig.add_trace(go.Scatter(x=t_full, y=y, mode="lines", name=f"{label} (traj)"))
 
+        if show in ("both", "nodes") and control_name in result.nodes:
+            data = result.nodes[control_name]
+            y = data if data.ndim == 1 else data[:, i]
+            fig.add_trace(go.Scatter(x=t_nodes, y=y, mode="markers", name=f"{label} (nodes)"))
+
+    fig.update_xaxes(title_text="Time (s)")
     return fig
 
 
-def _expanded_variable_names(states, controls):
-    names = []
+def plot_controls(
+    result: OptimizationResults,
+    control_names: list[str] | None = None,
+    include_private: bool = False,
+    show: str = "both",
```

Comment:
I am in favor of just removing this entirely as an argument and using a legendgroup to group all the nodal and propagation traces together. Then the user can use the legend to toggle the traces off or on.